integration_tests/suite/base/test_contexts.py

```diff
@@ -371,3 +375,50 @@ def test_bus_events(context):
     yield s.check_event, 'context_deleted', headers, confd.contexts(
         context['id']
     ).delete
+
+
+def test_create_contexts_parallel():
+    def create_context(responses, index):
+        response = confd.contexts.post(label='MyContext', wazo_tenant=DELETED_TENANT)
+        responses[index] = response
+
+    # create tenants (including DELETED_TENANT) in wazo-auth
+    BaseIntegrationTest.mock_auth.set_tenants(*ALL_TENANTS)
+
+    # check DELETED_TENANT does not exist in wazo-confd (not yet created)
+    response = confd.tenants(DELETED_TENANT).get(wazo_tenant=MAIN_TENANT)
+    response.assert_status(404)
+
+    threads = [None] * 5
+    responses = [None] * 5
+
+    for i in range(len(threads)):
+        threads[i] = Thread(target=create_context, args=(responses, i))
+        threads[i].start()
+
+    for i in range(len(threads)):
+        threads[i].join()
+
+    # check if tenant is created now (it has been created during the context creation)
+    response = confd.tenants(DELETED_TENANT).get(wazo_tenant=MAIN_TENANT)
+    response.assert_status(200)
+
+    # check there is no 500 error due to tenant creation conflict
+    assert_that(
+        responses,
+        not_(
+            has_item(
+                has_property("response", has_property("status_code", 500))
+                and has_property(
+                    "response",
+                    has_property(
+                        "text",
+                        matches_regexp(
+                            r"""Unexpected error: \(psycopg2.errors.UniqueViolation\) """
+                            + r"""duplicate key value violates unique constraint .*tenant_pkey.*"""
+                        ),
+                    ),
+                )
+            )
```

Comment:
IMHO, asserting for 500 status code is enough, we don't want *any* error to occur during this scenario.